File: 100 days of leetcode/Day20.py
class Solution:
    def compress(self, chars):
        if len(chars) <= 1:
            return len(chars)
        
        writeIndex = 0
        currentChar = chars[0]
        count = 1
        
        for i in range(1, len(chars)):
            if chars[i] == currentChar:
                count += 1
            else:
                chars[writeIndex] = currentChar
                writeIndex += 1
                if count > 1:
                    countStr = str(count)
                    for digit in countStr:
                        chars[writeIndex] = digit
                        writeIndex += 1
                
                currentChar = chars[i]
                count = 1
        
        chars[writeIndex] = currentChar
        writeIndex += 1
        if count > 1:
            countStr = str(count)
            for digit in countStr:
                chars[writeIndex] = digit
                writeIndex += 1
        
        return writeIndex

File: 100 days of leetcode/Day20.java
class Solution {
    public int compress(char[] chars) {
        int write_index = 0;
        int i = 0;
        
        while (i < chars.length) {
            char currentChar = chars[i];
            int count = 0;
            
            while (i < chars.length && chars[i] == currentChar) {
                i++;
                count++;
            }
            
            chars[write_index++] = currentChar;
            
            if (count > 1) {
                for (char digit : String.valueOf(count).toCharArray()) {
                    chars[write_index++] = digit;
                }
            }
        }
        
        return write_index;
    }
}
